### ISSUE TITLE
SpotBugs is disabled for the GUI module because of false positives

### ISSUE THREAD

**Issue by AdamCarroll** [Santulator/Santulator#3]
For JDK 11, it has been necessary to disable SpotBugs for the GUI module because of false positives relating to null checks.  The errors are:
* RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE: Redundant nullcheck of value known to be non-null
* RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE: Nullcheck of value previously dereferenced

**Comment by AdamCarroll** [comment]
This is probably caused by spotbugs/spotbugs#756

**Comment by AdamCarroll** [comment]
After reenabling SpotBugs, I now see the following warning:
~~~
> Task :gui:spotbugsTest
The following classes needed for analysis were missing:
  javafx.beans.property.SimpleBooleanProperty
  javafx.beans.property.SimpleObjectProperty
  javafx.beans.property.SimpleStringProperty
  javafx.scene.input.KeyCode
  javafx.scene.Node
  javafx.beans.property.BooleanProperty
  javafx.scene.control.TextInputControl
  javafx.beans.property.StringProperty
  javafx.beans.value.ObservableBooleanValue
  javafx.application.Platform
  javafx.collections.ObservableList
  javafx.scene.input.MouseButton
  javafx.application.Application
~~~

**Comment by AdamCarroll** [comment]
The project is now using v4.2.1 of SpotBugs.  This seems to improve matters but not fix the problem entirely.  It is now possible to remove one of the two exclusions from SpotBugs: 17a3fd0a123a6290277a33724a1c68254d1e2c46.  However, the following code still generates a false positive that means the second SpotBugs exclusion cannot be removed: https://github.com/Santulator/Santulator/blob/7e04627d4b6dbc7783e1ebb96e7d1fb46c9db981/gui/src/main/java/io/github/santulator/gui/common/BuildInfo.java#L26-L36

**Comment by AdamCarroll** [comment]
This problem is now fixed in the latest version of Spotbugs: 305d1b2d5a2b77dfdf12637e041e7257b2f2781b
